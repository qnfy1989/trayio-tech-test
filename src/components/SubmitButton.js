import React from 'react';
import styled from 'styled-components';

const ButtonWrapper = styled.button`
  align-self: flex-end;
  background-color: white;
  border-color: #4caf50;
  color: green;
  font-size: 13px;
  cursor: pointer;
  border-radius: 5px;
  margin-top: 20px;
  :hover {
    color: white;
    background-color: #4caf50;
  }
`;

function SubmitButton({children}) {
  return <ButtonWrapper type="submit">{children}</ButtonWrapper>;
}

export default SubmitButton;
