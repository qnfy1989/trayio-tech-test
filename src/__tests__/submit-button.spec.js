import React from 'react';
import {render, cleanup} from 'react-testing-library';
import SubmitButton from '../components/SubmitButton';

const setup = () => {
  const utils = render(<SubmitButton>Submit</SubmitButton>);
  const button = utils.getByText('Submit');
  return {
    button,
    ...utils,
  };
};

afterEach(cleanup);

test('it renders button with Submit value', () => {
  const {button} = setup();
  const firstChild = button.firstChild;
  expect(button.type).toEqual('submit');
  expect(firstChild).toMatchSnapshot();
});
