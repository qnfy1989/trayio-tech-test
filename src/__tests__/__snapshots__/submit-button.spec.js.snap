// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders button with Submit value 1`] = `Submit`;
